BridgeSDKTests/SBBParticipantManagerIntegrationTests.m

```diff
@@ -284,10 +287,50 @@ - (void)testSetSharingScope
 
 - (void)testSetExternalIdentifier
 {
+    NSMutableDictionary *headers = [NSMutableDictionary dictionary];
+    [gAdminAuthManager addAuthHeaderToHeaders:headers];
+
+    // Create an external identifier to test with
+    NSString *externalIdentifierFormat = @"test-external-identifier%@";
+    NSString *substudyId = @"test-external-id-substudy";
+    NSString *unique = [[NSProcessInfo processInfo] globallyUniqueString];
+    __block NSString *externalIdentifier = [NSString stringWithFormat:externalIdentifierFormat, unique];
+    NSDictionary *externalIdObject =
+    @{
+      @"identifier": externalIdentifier,
+      @"substudyId": substudyId,
+      @"type": @"ExternalIdentifier"
+      };
+    
+    XCTestExpectation *expectCreatedIdentifier = [self expectationWithDescription:@"created external identifier"];
+    id<SBBBridgeNetworkManagerProtocol> bridgeMan = SBBComponent(SBBBridgeNetworkManager);
+    [bridgeMan post:DEV_EXTERNAL_ID_API headers:headers parameters:externalIdObject completion:^(NSURLSessionTask *task, id responseObject, NSError *error) {
+        if (error) {
+            NSLog(@"Failed to create external ID %@\nError:%@\nResponse:%@", externalIdentifier, error, responseObject);
+            externalIdentifier = nil;
+        } else {
+            NSLog(@"Created test external ID %@", externalIdentifier);
+        }
+        
+        [expectCreatedIdentifier fulfill];
+    }];
+    
+    [self waitForExpectationsWithTimeout:5.0 handler:^(NSError *error) {
+        if (error) {
+            NSLog(@"Timeout creating external identifier %@: %@", externalIdentifier, error);
+            externalIdentifier = nil;
```

Comment:
I think how expectations work has changed and you need to include an assert here if there was an error.

Reply:
👍

Reply:
...actually remembered the reason I didn't do this: I want to make sure the test run doesn't halt prematurely so as much as possible of the test accounts etc. that it sets up in the test study can get cleaned up after each run even when there are failures. I still go in to the test study periodically to clean up stuff that got left behind when it hits an assert in the Bridge or iOS SDK, but it gets out of hand quickly if I don't let these continue after errors.